osu.Game/IO/StableStorage.cs

```diff
@@ -35,12 +35,13 @@ private string locateSongsDirectory()
         {
             var songsDirectoryPath = Path.Combine(BasePath, stable_default_songs_path);
 
-            var configFile = GetFiles(".", "osu!.*.cfg").FirstOrDefault();
+            var configFile = GetFiles(".", "osu!.*.cfg").SingleOrDefault();
```

Comment:
I don't think this is what you want. Now it will crash if there are more than one config files.

Reply:
I think an error is more appropriate here than guessing at random, but it shouldn't crash. Maybe this should be using the actual username instead of matching wildcards - that way there should never be a possibility of multiple matching files.

Reply:
Maybe, but it should fall back to using the first available. It is highly possible the username has changed, or the user has copied their files across a different OS install or PC.

Reply:
I'm not sure to follow why the detection should always fall back to the "oldest" of the config files available in the stable directory: 

if there happens to be more than one config file in the directory, and one of these contains the local username in the filename, this means that the user launched stable at least once after migrating it from another User / PC / OS installation and if he was using a custom beatmap dir, I guess he would have noticed by that point that the new config doesn't point to it and would probably have changed it?

That's my train of thoughts, not sure how applicable it is , though.

thoughts?

Reply:
Just pushed a commit that should make the file lookup behaviour match the behaviour you described above if I understood it correctly (f6d08f54e6e950f4784087012fe27cd3e566a22c). Please check it matches with the behaviour you meant.

Reply:
Your latest commit broke this... exact same thing again.